core/src/main/scala/org/bitcoins/core/psbt/PSBT.scala

```diff
@@ -896,6 +913,193 @@ case class InputPSBTMap(elements: Vector[InputPSBTRecord]) extends PSBTMap {
     getRecords(FinalizedScriptSigKeyId).nonEmpty || getRecords(
       FinalizedScriptWitnessKeyId).nonEmpty
 
+  /** Finalizes this input if possible, returning None if not */
+  def finalize(input: TransactionInput): Option[InputPSBTMap] = {
+    val scriptPubKeyToSatisfyOpt: Option[ScriptPubKey] = {
+      val witnessUTXOOpt = getRecords[WitnessUTXO](WitnessUTXOKeyId).headOption
+      val nonWitnessUTXOOpt =
+        getRecords[NonWitnessOrUnknownUTXO](NonWitnessUTXOKeyId).headOption
+
+      (witnessUTXOOpt, nonWitnessUTXOOpt) match {
+        case (None, None) => None
+        case (Some(witnessUtxo), _) =>
+          Some(witnessUtxo.spentWitnessTransaction.scriptPubKey)
+        case (_, Some(utxo)) =>
```

Comment:
This is missing the `case (Some(_), Some(_))`. It is concerning that there isn't a compiler warning here!

Reply:
Oh, it gets caught by the `_` in the above statements, is this the exact ordering you want? Seems a little dangerous.

Reply:
It should be fine, if you have a `WitnessUTXO` you can remove the `NonWitnessOrUnknownUTXO`

Reply:
what ben said ^ the spec says to remove `NonWitnessOrUnknownUTXO` if a `WitnessUTXO` record is present